### PR TITLE
Node 7.4.0

### PR DESCRIPTION
**Pull request by tbranyen**
Testing out build support on Travis... built locally, but hitting weird error:

```

#
# Fatal error in ../deps/v8/src/api.cc, line 1051
# Check failed: !value_obj->IsJSReceiver() || value_obj->IsTemplateInfo().
#

==== C stack trace ===============================

    /usr/bin/node(v8::base::debug::StackTrace::StackTrace()+0x16) [0x106bda6]
    /usr/bin/node(V8_Fatal+0xe8) [0x1069fe8]
    /usr/bin/node(v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)+0xff) [0x76094f]
    /home/tim/git/nodegit/nodegit/build/Release/nodegit.node(Wrapper::InitializeComponent(v8::Local<v8::Object>)+0x26d) [0x7fc0c190f58d]
    /home/tim/git/nodegit/nodegit/build/Release/nodegit.node(init+0x43) [0x7fc0c190d8a3]
    /usr/bin/node(node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)+0x365) [0xf51285]
    /usr/bin/node(v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))+0x12f) [0x78d6ef]
    /usr/bin/node() [0x7f2bb8]
    /usr/bin/node() [0x7f2f50]
    /usr/bin/node(v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)+0x2b) [0x7f312b]
    [0x3856731063a7]
```

### CODE REVIEW

**Comment by tbranyen** [comment]
Well that was unexpected... https://travis-ci.org/nodegit/nodegit/jobs/196246148

**Comment by johnhaley81** [comment]
Wat

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ env:
   matrix:
     - export NODE_VERSION="4" TARGET_ARCH="x64"
     - export NODE_VERSION="6.5" TARGET_ARCH="x64"
+    - export NODE_VERSION="7.4" TARGET_ARCH="x64"
 
 matrix:
   fast_finish: true
```
Comment (johnhaley81) [inline .travis.yml]: For 32-bit, it needs to be added to the matrix below as well.

Comment (tbranyen) [inline .travis.yml]: Okay cool, I also updated appveyor locally.

**Comment by johnhaley81** [comment]
Uhh sooo apparently we didn't need to modify OpenSSL after all for this bump. Which was the problem last time I tried this. Something must have changed between 7.0 and 7.4 or I'm losing my mind (also possible).

**Comment by tbranyen** [comment]
I also bumped node-pre-gyp and node-gyp in here, might have had something to do with it?

**Comment by johnhaley81** [comment]
Ahhh yeah that could be it. Good call.

**Comment by tbranyen** [comment]
Sweet, good to merge?

**Comment by johnhaley81** [comment]
:+1:

**Pull request merged**